src/js/jsx/shared/Datalist.jsx

```diff
@@ -48,10 +48,14 @@ define(function (require, exports, module) {
             live: React.PropTypes.bool,
             startFocused: React.PropTypes.bool,
             placeholderText: React.PropTypes.string,
-            // option to render when there are no other valid options to display
-            placeholderOption: React.PropTypes.instanceOf(Immutable.Iterable),
-            useAutofill: React.PropTypes.bool, // displays a suggested option next to the inputted text
-            neverSelectAllInput: React.PropTypes.bool // if true, will not highlight input text on commit
+            // Option to render when there are no other valid options to display
```

Comment:
These are really useful comments. At some point, since you've spent so much time inside `Datalist`, would you mind trying to provide the same level of documentation for all the state properties as well?